packages/blade/src/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,204 @@
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable @typescript-eslint/no-shadow */
+/* eslint-disable @typescript-eslint/prefer-nullish-coalescing */
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+import { isUndefined } from 'lodash';
+import React from 'react';
+import { isEmpty } from '../../utils';
+import { useCheckboxGroupContext } from './CheckboxGroup/CheckboxGroupContext';
+import { CheckboxHintText } from './CheckboxHintText';
+import { CheckboxIcon } from './CheckboxIcon';
+import { CheckboxInput } from './CheckboxInput';
+import { CheckboxLabel } from './CheckboxLabel';
+import { CheckboxLabelText } from './CheckboxLabelText';
+import { useCheckbox } from './useCheckbox';
+import { Wrapper } from './Wrapper';
+
+type CheckboxProps = {
+  /**
+   * If `true`, sets the checkbox will be checked.
+   * Use `onChange` to update its value
+   *
+   * @default false
+   */
+  isChecked?: boolean;
+  /**
+   * If `true`, the checkbox will be initially checked.
+   *
+   * @default false
+   */
+  defaultChecked?: boolean;
+  /**
+   * The callback invoked when the checked state of the `Checkbox` changes.
+   */
+  onChange?: (isChecked: boolean) => void;
+  /**
+   * Sets the label text of the checkbox
+   */
+  children: string;
+  /**
+   * Help text of the checkbox
+   */
+  helpText?: string;
+  /**
+   * Error text of the checkbox
+   *
+   * Renders when `validationState` is set to 'error'
+   *
+   * Overrides helpText
+   */
+  errorText?: string;
+  /**
+   * If `true`, the checkbox will be indeterminate.
+   * This does not modify the isChecked property.
```

Comment:
so if onchange is clicked and isIndeterminate is true then also it'll return true?

Reply:
Yes, indeterminate you can think of it as a fake state. Since users need to handle the behaviour and logic of actual indeterminate checkbox. And both isChecked & indeterminate are independent.